### PR TITLE
Fix: External detectors couldn't run a match case on ExtractedXXX::from(YY).extracted because AST node structs weren't available

### PR DESCRIPTION
**Pull request by TilakMaddy**
Solution: Re-export core ast into aderyn_driver for external detectors for ease of access.
An example detector that requires this ast node matching has been added to `bot_example`  to demonstrate the same.

Other solution:
Of course, the users can always cargo add `aderyn_core` for doing that but it makes it more convenient if we re-export this one more module in aderyn_driver via `core_ast`

### CODE REVIEW

File: bot_example/src/delegate_call_in_loop.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    found_instances: BTreeMap<(String, usize), String>,
+}
+
+impl Detector for DelegateCallInLoopDetector {
+    fn detect(&mut self, loader: &ContextLoader) -> Result<bool, Box<dyn Error>> {
+        let mut member_accesses: Vec<MemberAccess> = vec![];
+
+        // Get all delegatecall member accesses inside for statements
+        member_accesses.extend(loader.for_statements.keys().flat_map(|statement| {
+            ExtractMemberAccesses::from(statement)
```
Comment (alexroan) [inline bot_example/src/delegate_call_in_loop.rs:22]: Does the parent finder work here? `source_unit_of`, `contract_definition_of`, etc?

**Review by alexroan — COMMENTED** [review]
Great catch, @TilakMaddy , one question.

File: bot_example/src/delegate_call_in_loop.rs (hunk 1 above)
Comment (TilakMaddy) [reply bot_example/src/delegate_call_in_loop.rs:22]: Yes ! `GetParent::XX` works - just tested it! 

All the stuff from the `browser` module is visible : )

**Review by alexroan — APPROVED** [review]
@TilakMaddy happy to merge once conflicts are resolved :)

**Review by alexroan — APPROVED** [review]

**Pull request merged**